src/AppBundle/Controller/TodoController.php

```diff
@@ -0,0 +1,242 @@
+            ->getRepository('AppBundle:Todo')
+            ->find($id);
+
+        if (!$item) {
+            return new JsonResponse('Not Found', 404);
+        }
+
+        //if we are forbidden (we are authorized)
+        if ($item->getUserId() != $this->getUser()->getIdString()) {
+            return new JsonResponse('Forbidden', 403);
+        }
+
+        return new JsonResponse($item->serialize());
+    }
+
+    /**
+     * @Route("/todo/{id}", name="delete_todo_id")
+     * @Method({"DELETE"})
+     *
+     * @param string $id
+     * @param Request $request
+     * @return JsonResponse
+     */
+    public function deleteTodoIdAction($id, Request $request)
+    {
+        $item = $this->get('doctrine_mongodb')
+            ->getRepository('AppBundle:Todo')
+            ->find($id);
+
+        if (!$item) {
+            return new JsonResponse('Not Found', 404);
+        }
+
+        //if we are forbidden (we are authorized)
+        if ($item->getUserId() != $this->getUser()->getIdString()) {
+            return new JsonResponse('Forbidden', 403);
+        }
+
+        $this->dm()->remove($item);
+        $this->dm()->flush();
+
+        return new JsonResponse('Successfully deleted ' . $id);
+    }
+
+    /**
+     * @Route("/todo/{id}", name="put_todo_id")
+     * @Method({"PUT"})
+     *
+     * @param string $id
+     * @param Request $request
+     * @return JsonResponse
+     */
+    public function putTodoIdAction($id, Request $request)
+    {
+        $item = $this->get('doctrine_mongodb')
+            ->getRepository('AppBundle:Todo')
+            ->find($id);
+
+        if (!$item) {
+            return new JsonResponse('Not Found', 404);
+        }
+
+        //if we are forbidden (we are authorized)
+        if ($item->getUserId() != $this->getUser()->getIdString()) {
+            return new JsonResponse('Forbidden', 403);
+        }
+
+        $body = array();
+        try {
+            $body = $this->bodyAsJSON($request);
+        } catch (JSONControllerException $e) {
+            return new JsonResponse('' . $e, 400);
+        }
+
+        $title = null;
+        $completed = null;
+
+        foreach ($body as $key => $value) {
+            if ($key == 'title') {
+                if (is_string($value)) {
+                    $title = $value;
+                    $item->setTitle($title);
+                } else
+                    return new JsonResponse('Bad Request: The body parameter "title" was not type "string".', 400);
+
+            } else if ($key == 'completed') {
+                if (is_bool($value) || is_int($value)) {
+                    $completed = ($value === true || $value === 1);
+                    $item->setCompleted($completed);
+                } else
+                    return new JsonResponse('Bad Request: The body parameter "completed" was not type "bool".', 400);
+            }
+        }
+
+        $this->dm()->flush();
+
+        if ($title == null)
```

Comment:
I had it there for debug, since testing was done black box style it was easier then logging it internally.